From 3f35f4a9243b860f0ae92728a5aadcda1bf1568f Mon Sep 17 00:00:00 2001
From: "Ivan.liu" <xiaowen.liu@nxp.com>
Date: Mon, 22 Oct 2018 09:19:06 +0800
Subject: [PATCH 4905/5242] MA-13121-2 [#imx-1221] Fix kernel panic after
 reduce gpu reserved memory size.

commit  1300ab1c6b367ce3bd4ff7f231aba150f6aa9139 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix uint64_t to pointer conversion issue on 32bit platform.

Signed-off-by: Ivan.liu <xiaowen.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../hal/os/linux/kernel/gc_hal_kernel_drm.c        |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_drm.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_drm.c
index 157e425..a047d4e 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_drm.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_drm.c
@@ -582,7 +582,7 @@ static int viv_ioctl_gem_attach_aux(struct drm_device *drm, void *data,
         gcmkONERROR(gckDEVICE_Dispatch(gal_dev->device, &iface));
 
         /* Fill tile status node with tileStatusFiller. */
-        entry = (char __user*)iface.u.LockVideoMemory.memory;
+        entry = (char __user*)(uintptr_t)iface.u.LockVideoMemory.memory;
         for (num=0; num<gem_ts_obj->size; num++) {
             put_user(tileStatusFiller, entry+num);
         }
-- 
1.7.9.5

